### ISSUE TITLE
Numbers in scientific notation not recognized as floats in 3.0

### ISSUE THREAD

**Issue by hamoid** [processing/processing#4190]
This used to be possible: 

    rotate(90e1);

In 3.0 it complains that 90e1 is not a float. Also 

    rotate(float(90e1)); 

does not work, but 

    rotate((float)90e1); 

does.

**Comment by djokjulapfe** [comment]
You have to put 90.0e1. Notice de .0 I added. 90e1 creates a double, and rotate works only with floats.

**Comment by hamoid** [comment]
Good to know. But my point is that this behaviour changed. It used to work without the .0 in previous versions, and old programs I wrote stopped working. Maybe it was mentioned and I missed it.

**Comment by Akarshit** [comment]
From what i could see, some of the conversion functions have been remove like boolean(), byte(), char(), float(), int(). This is the reason for  

`rotate(float(90e1));`  

to not work. But they are still given as a part of the reference in 3.0 (like for float() [here](https://www.processing.org/reference/floatconvert_.html)).

**Comment by hamoid** [comment]
@Akarshit are you sure they have been removed? They work all right in my 3.0.

    println(float(22000)); // works, float() is there
    println((float)22e3); // works, but 22e3 interpreted as double, converted to float
    println(float(22e3)); // not  working. can't convert double to float with float()

I'm wondering, though, since I see no mention of the scientific notation in the documentation, even if the behavior changed without notice, if this issue can be closed? Nothing was ever promised regarding scientific notation, right? Only some of my http://funprogramming.org/p5tweets/ broke, which is to be expected anyway :)

**Comment by Akarshit** [comment]
@hamoid Sorry my bad, i got confused somewhere.

**Comment by GoToLoop** [comment]
@hamoid , `22e3` isn't interpreted as `double` in the PDE. We have to suffix it w/ `d` or `D` for it:

```Java
float f1 = 22.e3;  // works
float f2 = 22.e3d; // cannot convert from double to float
```

PDE's  pre-processor visits each literal in the ".pde" code and suffix every 1 which got a `.` or an `e` in it.
Of course, as long as it doesn't have 1 already.  ;-)

**Comment by GoToLoop** [comment]
> println(float(22e3)); // not  working. can't convert double to float with float()

The actual compile error is: *"The method parseFloat(int) in the type PApplet is not applicable for the argument (float)"*

**Comment by JakubValtar** [comment]
Fixed in #4440
